### PR TITLE
indloggetudlogget funktionalitet DDFSOEG 178

### PR DESCRIPTION
**Pull request by spaceo**
#### Link to issue

https://reload.atlassian.net/browse/DDFSOEG-178

#### Description
This PR introduces an authorization endpoint that the react apps can use for getting redirected to Adgangsplatformen.

#### Checklist

- [ ] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [ ] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [ ] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

### CODE REVIEW

File: web/modules/custom/dpl_login/src/Controller/DplLoginController.php [hunk 1]
```diff
@@ -126,4 +147,30 @@ public function logout() {
...
+      $_SESSION['openid_connect_destination'] = $current_path;
+    }
+
+    /** @var \Drupal\openid_connect\Plugin\OpenIDConnectClientInterface $client */
+    $client = $this->pluginManager->createInstance(
+      self::CLIENT_NAME,
+      $this->settings
+    );
+
+    return new TrustedRedirectResponse($client->getAuthorizationEndpointUrl("openid email profile"));
```
Comment (kasperg) [inline web/modules/custom/dpl_login/src/Controller/DplLoginController.php:173]: Consider if adding `getAuthorizationEndpointUrl()` is the right approach for maintainability going forward.

I have looked into [the patch](https://www.drupal.org/files/issues/2022-08-26/openid_connect-auth-endpoint.patch) and this code that uses the patch.

As far as I can tell it seems as if the patch creates a method in the `OpenIDConnectClientBase` for generating an url which this method then passes to a `TrustedRedirectResponse`. Without the patch [the `authorize` method the the class seems to return the same url wrapped in a `TrustedRedirectResponse`](https://git.drupalcode.org/project/openid_connect/-/blob/8.x-1.x/src/Plugin/OpenIDConnectClientBase.php#L256). Why not just call `$client->authorize()` and return the result.

[This approach is used to solve a similar problem in Gruppeweb](https://github.com/reload/gruppeweb/blob/main/web/modules/custom/gruppeweb_login/src/Controller/Login.php#L99-L123).

**Review by kasperg — CHANGES_REQUESTED** [review]
A couple of things need changing here and a few things for you to consider.

I have only reviewed parts of the PR not already included in #120 .

Also remember to remove commits with custom builds before merging.

File: composer.json [hunk 2]
```diff
@@ -147,6 +147,9 @@
             },
             "drupal/jsonlog": {
                 "3251587: Change logging from stdout to stderr": "https://www.drupal.org/files/issues/2021-11-29/jsonlog-change-stdout-to-stderr-3251587-3.patch"
+            },
+            "drupal/openid_connect": {
+                "Expose authorization url endpoint": "https://www.drupal.org/files/issues/2022-08-26/openid_connect-auth-endpoint.patch"
```
[the remark this answers is not in the document]
Comment (spaceo) [reply composer.json:152]: removing the patch...

File: web/modules/custom/dpl_login/dpl_login.routing.yml [hunk 3]
```diff
@@ -6,3 +6,13 @@ dpl_login.logout:
   requirements:
     _user_is_logged_in: 'TRUE'
   no_cache: TRUE
+
+dpl_login.authorize_from_app:
+  path: '/dpl-login/authorization-url'
+  defaults:
+    _controller: '\Drupal\dpl_login\Controller\DplLoginController::authorizeFromApp'
+    _title: 'Authorize From App'
```
[the remark this answers is not in the document]
Comment (spaceo) [reply web/modules/custom/dpl_login/dpl_login.routing.yml:14]: done

**Review by cableman — APPROVED** [review]

**Pull request merged**